btk/catalog.py

```diff
@@ -78,26 +79,33 @@ def _prepare_table(self):
 
 
 class CosmosCatalog(Catalog):
+
+    def __init__(self, catalog, galsim_catalog, verbose=False):
+        super().__init__(catalog, verbose=verbose)
+        self.galsim_catalog = galsim_catalog
+    
     @classmethod
     def from_file(cls, catalog_files, verbose=False):
         """
         Args:
             catalog_files: list of galsim cataolgs
         """
-        catalog_coord = astropy.table.Table.read(catalog_files[0], "fits")
-        catalog_fit = astropy.table.Table.read(catalog_files[1], "fits")
+        catalog_coord = astropy.table.Table.read(catalog_files[0])
+        catalog_fit = astropy.table.Table.read(catalog_files[1])
         catalog = astropy.table.hstack([catalog_coord, catalog_fit])
 
-        return cls(catalog, verbose=verbose)
+        galsim_catalog = galsim.COSMOSCatalog(catalog_files[0])
+        return cls(catalog, galsim_catalog, verbose=verbose)
 
     def _prepare_table(self):
         table = deepcopy(self._raw_catalog)
         # make notation for 'ra' and 'dec' standard across code.
         table.rename_column("RA", "ra")
         table.rename_column("DEC", "dec")
         table.rename_column("MAG", "ref_mag")
-        index = np.where(t["IDENT_1"] == self._raw_catalog["IDENT_1"] for t in table)
-
+        #index = np.where(t["IDENT_1"] == self._raw_catalog["IDENT_1"] for t in table)
```

Comment:
remove this commented out version?

Reply:
> Isn't the size difference due to the fact that there are two files imported for the catalog ? Try removing the catalog_fit line ; does it make a difference ? Otherwise, the process seems fine to me ; if we have a separate variable I am not sure that we really need _raw_catalog anymore.

The size of `catalog_coord` and `catalog_fit` are the same (87798) while the galsim catalog has 81499 entries...
If we remove `_raw_catalog`, we would have to have it as a parameter of `_prepare_table`, isn't it ? But I think we should still read `catalog_coord` and `catalog_fit` at some point because I am not sure we can retrieve the necessary information in the galsim catalog.

> Regarding the fluxes, it could be as simple at that. Did you draw some example of blends and do the fluxes look right to you ? You can put some here if you want.

It seems ok but I think we should solve the problem of shifts to assess this. Currently shifts are not used. I guess we should use `entry['ra']` and `entry['dec']` in `render_single`. 
 
> SIde note : you may want to use the pre-commit to pass the linting on the first try :)

Indeed. I just switched from a fork to a clone and forgot to install it :)